### PR TITLE
Http signature middleware

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: lib/httpSignatureStrategy.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    if(err.name === 'NotFound') {
+      // fallback to basic id
+      return {
+        id: owner,
+        identity: {
+          '@context': bedrock.config.constants.IDENTITY_CONTEXT_V1_URL,
+          id: owner
+        }
+      };
+    }
```
Comment (dlongley) [inline lib/httpSignatureStrategy.js:70]: Do we need to `throw err` when it's not a `NotFound` error?

File: lib/httpSignatureStrategy.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    wrap: (url, callback) => bedrock.jsonld.documentLoader(url, callback),
+  });
+  jsigs = jsigs();
+  jsigs.use('jsonld', jsonld);
+  api.strategy.use('jsigs', jsigs);
+});
+
+function _getKey(publicKey) {
+  return new Promise((resolve, reject) => {
+    brKey.getPublicKey(publicKey, (err, key) => {
```
Comment (dlongley) [inline lib/httpSignatureStrategy.js]: Could we have used `promisify` here too?

File: lib/httpSignatureStrategy.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+bedrock.events.on('bedrock.init', () => {
+  // get configured lib instances
+  jsonld = jsonld();
+  didio = didio();
+  didio.use('jsonld', jsonld);
+  jsonld.documentLoader = didio.createDocumentLoader({
+    baseUrl: bedrock.config.passport.strategies.did.didio.baseUrl,
+    wrap: (url, callback) => bedrock.jsonld.documentLoader(url, callback),
+  });
```
Comment (dlongley) [inline lib/httpSignatureStrategy.js:32]: I guess this stuff is fine for now to go into `master` ... but we will need to rev it almost right away. I don't think there's a good reason to release a new version of bedrock-passport using the old did-io unless it's totally non-breaking and helps us make progress somewhere else. I think we should be getting pretty close to getting the new did-io.

**Comment by mattcollier** [comment]
@dlongley ready for another pass.

**Review by dlongley — APPROVED** [review]
LGTM, other than my caveat that I am skeptical about doing a new release before:

1. The new did-io stuff lands.
2. The bedrock-account stuff lands.

**Pull request merged**